workflow/artifacts/s3/s3.go

```diff
@@ -103,6 +105,27 @@ func (s3Driver *S3ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifac
 				log.Warnf("Failed to test if %s is a directory: %v", path, err)
 				return false, nil
 			}
+
+			if outputArtifact.S3.CreateBucketIfNotPresent.Enabled {
+				bucketExists, err := s3cli.BucketExists(outputArtifact.S3.Bucket)
+				if err != nil {
+					log.Warnf("Failed to check if bucket exists: %v", outputArtifact.S3.Bucket)
```

Comment:
why not return error here?

Reply:
@alexec I tried to follow the same pattern here as with other errors. Since it's wrapped in an `ExponentialBackoff` no errors are returned inside the function as it would cause the error to be returned after a failure after the first attempt.

Happy to change it so it fails after the first attempt, but it would be an exception to this code block